spec.bs

```diff
@@ -257,14 +257,191 @@ dictionary AuctionAdConfig {
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer signals=][|buyer|] to
+    |signalsString|.
+1. If |config|["{{AuctionAdConfig/perBuyerTimeouts}}"] [=map/exists=], [=map/for each=] |key| →
+  |value| of |config|["{{AuctionAdConfig/perBuyerTimeouts}}"]:
+  1. If |key| equals to "*", then set |auctionConfig|'s [=auction config/all buyers timeout=]
+    to |value|, and [=iteration/continue=].
+  1. Let |buyer| be parsing |key| to an [=url/origin=]. If it fails, [=exception/throw=] a
+    {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer timeouts=][|buyer|] to |value|.
+1. If |config|["{{AuctionAdConfig/perBuyerExperimentGroupIds}}"] [=map/exists=], [=map/for each=]
+  |key| → |value| of |config|["{{AuctionAdConfig/perBuyerExperimentGroupIds}}"]:
+  1. If |key| equals to "*", then set |auctionConfig|'s |auctionConfig|'s
+    [=auction config/all buyer experiment group id=] to |value|, and [=iteration/continue=].
+  1. Let |buyer| be parsing |key| to an [=url/origin=]. If it fails, [=exception/throw=] a
+    {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer experiment group ids=][|buyer|] to
+    |value|.
+1. If |config|["{{AuctionAdConfig/perBuyerGroupLimits}}"] [=map/exists=], [=map/for each=]
+  |key| → |value| of |config|["{{AuctionAdConfig/perBuyerGroupLimits}}"]:
+  1. If |value| is less or equal to 0, [=exception/throw=] a {{TypeError}}.
+  1. If |key| equals to "*", then set |auctionConfig|'s [=auction config/all buyers group limit=]
+    to |value|, and [=iteration/continue=].
+  1. Let |buyer| be parsing |key| to an [=url/origin=]. If it fails, [=exception/throw=] a
+    {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer group limits=][|buyer|] to |value|.
+1. If |config|["{{AuctionAdConfig/perBuyerPrioritySignals}}"] [=map/exists=], [=map/for each=]
+  |key| → |value| of |config|["{{AuctionAdConfig/perBuyerPrioritySignals}}"]:
+  1. Let |signals| be an [=ordered map=] whose [=map/key|keys=] are [=strings=] and whose
+    [=map/value|values=] are double.
+  1. [=map/for each=] |k| → |v| of |value|:
+    1. If |k| [=string/starts with=] "browserSignals.", [=exception/throw=] a {{TypeError}}.
+    1. Set |signals|[|k|] to |v|.
+  1. If |key| equals to "*", then set |auctionConfig|'s
+    [=auction config/all buyers priority signals=] to |value|, and [=iteration/continue=].
+  1. Let |buyer| be parsing |key| to an [=url/origin=]. If it fails, [=exception/throw=] a
+    {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer priority signals=][|buyer|] to
+    |signals|.
+1. If |config|["{{AuctionAdConfig/sellerTimeout}}"] [=map/exists=], set |auctionConfig|'s
+  [=auction config/seller timeout=] to |config|["{{AuctionAdConfig/sellerTimeout}}"].
+1. If |config|["{{AuctionAdConfig/componentAuctions}}"] [=map/exists=], [=list/for each=]
+  |component| in |config|["{{AuctionAdConfig/componentAuctions}}"]:

```

Comment:
Set its [default](https://webidl.spec.whatwg.org/#dfn-dictionary-member-default-value) to `= []`, and you won't need to check for existence here:
```suggestion
1. [=list/For each=] |component| in |config|["{{AuctionAdConfig/componentAuctions}}"]:
```